Lib/_collections_abc.py

```diff
@@ -1064,8 +1064,27 @@ def count(self, value):
 Sequence.register(range)
 Sequence.register(memoryview)
 
-
-class ByteString(Sequence):
+class _DeprecateByteStringMeta(ABCMeta):
+    def __new__(cls, name, bases, namespace, **kwargs):
+        if name != "ByteString":
+            import warnings
+
+            warnings._deprecated(
```

Comment:
Have you checked whether these two warnings get raised in the right place? They're raised in somewhat unusual places, so maybe the stacklevel needs to be incremented.

Reply:
Yeah, I checked, it looked right to me!